src/rail_cmd.cpp

```diff
@@ -263,11 +263,9 @@ static CommandCost CheckTrackCombination(TileIndex tile, TrackBits to_build, uin
 	}
 
 	/* Let's see if we may build this */
-	if ((flags & DC_NO_RAIL_OVERLAP) || HasSignals(tile)) {
-		/* If we are not allowed to overlap (flag is on for ai companies or we have
-		 * signals on the tile), check that */
+	if ((flags & DC_NO_RAIL_OVERLAP)) { /* DC_NO_RAIL_OVERLAP flag is on for ai companies */
 		if (future != TRACK_BIT_HORZ && future != TRACK_BIT_VERT) {
```

Comment:
We recently changed this piece of code, as `DC_NO_RAIL_OVERLAP` was unused. You can just remove this whole block now :) We made it easier for you :D